helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -1036,6 +1037,46 @@ public static void purgeExpiredJobs(String workflow, WorkflowConfig workflowConf
     setNextJobPurgeTime(workflow, currentTime, purgeInterval, rebalanceScheduler, manager);
   }
 
+  /**
+   * The function that loops through the all existed workflow contexts and removes IdealState and
+   * Workflow Context if Workflow Config is missing.
+   * @param dataProvider
+   * @param manager
+   */
+  public static void workflowGarbageCollection(WorkflowControllerDataProvider dataProvider,
+      final HelixManager manager) {
+    // Garbage collections for conditions where workflow context is existed but config is missing.
+    Map<String, ZNRecord> contexts = dataProvider.getContexts();
+    for (Map.Entry<String, ZNRecord> entry : contexts.entrySet()) {
```

Comment:
When you do this, you'd want to take a snapshot of the entries because this might cause a ConcurrentModificationException.

Reply:
Very good point. Fixed. Thanks.